lib/alexa.js

```diff
@@ -1,418 +1,573 @@
+  });
+
+  Object.defineProperty(handler, '_context', {
+    value: context,
+    writable: false,
+  });
+
+  Object.defineProperty(handler, '_callback', {
+    value: callback,
+    writable: false,
+  });
+
+  Object.defineProperty(handler, 'state', {
+    value: null,
+    writable: true,
+    configurable: true,
+  });
+
+  Object.defineProperty(handler, 'appId', {
+    value: null,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'response', {
+    value: {},
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'dynamoDBTableName', {
+    value: null,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'saveBeforeResponse', {
+    value: false,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'i18n', {
+    value: i18n,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'locale', {
+    value: undefined,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'resources', {
+    value: undefined,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'registerHandlers', {
+    value: function registerHandlers(...args) {
+      RegisterHandlers.apply(handler, args);
+    },
+    writable: false,
+  });
+
+  Object.defineProperty(handler, 'execute', {
+    value: function execute() {
+      HandleLambdaEvent.call(handler);
+    },
+    writable: false,
+  });
+
+  Object.defineProperty(handler, 't', {
+    value: function t(...args) {
+      return this.i18n.t(...args);
+    },
+  });
+
+  handler.registerHandlers(responseHandlers);
+
+  return handler;
 }
 
+/**
+ * HandleLambdaEvent - Set up i18next translation backend, then call ValidateRequest.
+ *
+ * NB: this keyword must be bound to handler object.
+ *
+ * @return {undefined}
+ */
 function HandleLambdaEvent() {
-    this.locale = this._event.request.locale;
-    if(this.resources) {
-        this.i18n.use(sprintf).init({
-            overloadTranslationOptionHandler: sprintf.overloadTranslationOptionHandler,
-            returnObjects: true,
-            lng: this.locale,
-            resources: this.resources
-        }, (err, t) => {
-            if(err) {
-                throw new Error('Error initializing i18next: ' + err);
-            }
-            ValidateRequest.call(this);
-        });
-    } else {
-        ValidateRequest.call(this);
-    }
+  this.locale = this._event.request.locale;
+  if (this.resources) {
+    this.i18n.use(sprintf).init({
+      overloadTranslationOptionHandler: sprintf.overloadTranslationOptionHandler,
+      returnObjects: true,
+      lng: this.locale,
+      resources: this.resources,
+    }, (err) => {
+      if (err) {
+        throw new Error(`Error initializing i18next: ${err}`);
+      }
+      ValidateRequest.call(this);
+    });
+  } else {
+    ValidateRequest.call(this);
+  }
 }
 
+/**
+ * ValidateRequest - Check that App IDs match, get attributes from DynamoDB, then call EmitEvent.
+ *
+ * NB: this keyword must be bound to handler object.
```

Comment:
https://en.wikipedia.org/wiki/Nota_bene - I just intended that to be a note to the user, and NB is flagged in many IDEs and text editors. Is there a better way of achieving this?